operator/src/main/java/oracle/kubernetes/operator/calls/AsyncRequestStep.java

```diff
@@ -377,14 +377,18 @@ public NextAction doPotentialRetry(Step conflictStep, Packet packet, int statusC
         }
 
         NextAction na = new NextAction();
-        if (statusCode == 0 && retryCount <= maxRetryCount) {
-          na.invoke(Optional.ofNullable(conflictStep).orElse(retryStep), packet);
-        } else {
-          LOGGER.finer(MessageKeys.ASYNC_RETRY, identityHash(), String.valueOf(waitTime));
-          na.delay(retryStep, packet, waitTime, TimeUnit.MILLISECONDS);
+        if (retryCount <= maxRetryCount) {
+          if (statusCode == 0) {
+            na.invoke(retryStep, packet);
+          } else {
+            LOGGER.finer(MessageKeys.ASYNC_RETRY, identityHash(), String.valueOf(waitTime));
+            na.delay(retryStep, packet, waitTime, TimeUnit.MILLISECONDS);
+          }
+          return na;
         }
-        return na;
-      } else if (statusCode == 409 /* Conflict */ && conflictStep != null) {
+      } else if (statusCode == 409 /* Conflict */
```

Comment:
This conditional has now become complex enough that it should probably be broken out into another method. I would recommend changing it to```
    } else if (retriesLeft() && isRestartableConflict(conflictStep, status)) {```

where you also have:```

    private isRestartableConflict(Step conflictStep, int status) {
      return conflictStep != null && status == 409;
    }```

This would also allow you to move the comment explain optimistic locking failures to just above the new method

Reply:
I have made the suggested changes, please review, Thanks.